diff --git a/lib/ApiExtractorPlugin.js b/lib/ApiExtractorPlugin.js
index 74c10e645c59610fe79950f4ac57dd170879da2c..549672cec812bcf13694dd0ef9f6357bb2a1abc3 100644
--- a/lib/ApiExtractorPlugin.js
+++ b/lib/ApiExtractorPlugin.js
@@ -61,8 +61,8 @@ class ApiExtractorPlugin {
                 this._apiExtractorConfigurationFilePath =
                     await heftConfiguration.rigConfig.tryResolveConfigFilePathAsync(LEGACY_EXTRACTOR_CONFIG_RELATIVE_PATH);
                 if (this._apiExtractorConfigurationFilePath !== undefined) {
-                    taskSession.logger.emitWarning(new Error(`The "${LEGACY_EXTRACTOR_CONFIG_RELATIVE_PATH}" configuration file path is not supported ` +
-                        `in Heft. Please move it to "${EXTRACTOR_CONFIG_RELATIVE_PATH}".`));
+                    //taskSession.logger.emitWarning(new Error(`The "${LEGACY_EXTRACTOR_CONFIG_RELATIVE_PATH}" configuration file path is not supported ` +
+                    //    `in Heft. Please move it to "${EXTRACTOR_CONFIG_RELATIVE_PATH}".`));
                 }
             }
         }
