diff --git a/Makefile b/Makefile
index 6a64e164..60ffef95 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 GOFLAGS := -tags load_wgnt_from_rsrc -ldflags="-H windowsgui -s -w" -trimpath -buildvcs=false -v
 export GOOS := windows
-export PATH := $(CURDIR)/.deps/go/bin:$(PATH)
+# export PATH := $(CURDIR)/.deps/go/bin:$(PATH)
 
 VERSION := $(shell sed -n 's/^\s*Number\s*=\s*"\([0-9.]\+\)"$$/\1/p' version/version.go)
 empty :=
